app/settings/account/-components/connected-emails/component.ts

```diff
@@ -118,23 +119,31 @@ export default class ConnectedEmails extends Component.extend({
     }
 
     @action
-    onSave(email: UserEmail) {
-        if (email.emailAddress) {
-            this.set('lastUserEmail', email.emailAddress);
+    onSave(changeset: ChangesetDef) {
+        if (changeset.get('emailAddress')) {
+            this.set('lastUserEmail', changeset.get('emailAddress'));
             this.set('showAddModal', true);
             this.reloadUnconfirmedList();
 
             return this.toast.success(this.i18n.t('settings.account.connected_emails.save_success'));
         }
     }
     @action
-    onError(userEmail: UserEmail, e: DS.AdapterError | Error) {
+    onError(changeset: ChangesetDef, e: DS.AdapterError | Error) {
```

Comment:
Aren't these parameters backward? should be `onError(e, changeset)`

Reply:
I mean, you would think. I look forward to a day when templates will work with typescript.